File: Problem set 1/cash.c
// Greedy algorithm

#include <math.h>
#include <stdio.h>
#include <cs50.h>

int get_change(void);
void min_coins(int);

int main(void)
{
    int cents = get_change();
    min_coins(cents);
}

int get_change(void)
{
    float dollars;
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0);

    int cents = round(dollars * 100);
    return cents;
}

void min_coins(int cents)
{
    int coins = 0;

    if (cents == 0)
    {
        coins = 0;
    }
    else
    {
        while (cents > 0)
        {
            while (cents >= 25)
            {
                coins += cents / 25;
                cents = cents % 25;
            }
            while (cents >= 10)
            {
                coins += cents / 10;
                cents = cents % 10;
            }
            while (cents >= 5)
            {
                coins += cents / 5;
                cents = cents % 5;
            }
            while (cents >= 1)
            {
                coins += cents / 1;
                cents = cents % 1;
            }
        }
    }
    printf("%i\n", coins);    
}

File: problem_set_6/cash.py
from cs50 import get_float

while True:
    change = get_float("Change owed: ") * 100
    if change >= 0:
        break

coin_count = 0

while change > 0:
    while change >= 25:
        coin_count += change // 25
        change %= 25
    while change >= 10:
        coin_count += change // 10
        change %= 10
    while change >= 5:
        coin_count += change // 5
        change %= 5
    while change >= 1:
        coin_count += change // 1
        change %= 1

print(int(coin_count))
